### PR TITLE
python3Packages.langgraph* 20241006 update

### PR DESCRIPTION
**Pull request by sarahec**
## Changes

`python3Packages.langgraph*`:

* `langgraph`: 0.2.21 -> 0.2.34
* `langgraph-sdk`: 0.1.30 -> 0.1.32
* `langgraph-checkpoint`: 1.0.9 -> 2.0.0
* `langgraph-checkpoint-postgres`: 1.0.6 -> 2.0.0
* `langgraph-checkpoint-sqlite`: 1.0.3 -> 2.0.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 346948` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.langgraph</li>
    <li>python311Packages.langgraph-checkpoint</li>
    <li>python311Packages.langgraph-checkpoint-postgres</li>
    <li>python311Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python311Packages.langgraph-checkpoint-sqlite</li>
    <li>python311Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python311Packages.langgraph-checkpoint.dist</li>
    <li>python311Packages.langgraph-sdk</li>
    <li>python311Packages.langgraph-sdk.dist</li>
    <li>python311Packages.langgraph.dist</li>
    <li>python312Packages.langgraph</li>
    <li>python312Packages.langgraph-checkpoint</li>
    <li>python312Packages.langgraph-checkpoint-postgres</li>
    <li>python312Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python312Packages.langgraph-checkpoint-sqlite</li>
    <li>python312Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python312Packages.langgraph-checkpoint.dist</li>
    <li>python312Packages.langgraph-sdk</li>
    <li>python312Packages.langgraph-sdk.dist</li>
    <li>python312Packages.langgraph.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/langgraph-checkpoint/default.nix [hunk 1]
```diff
@@ -31,11 +33,16 @@ buildPythonPackage rec {
 
   dependencies = [ langchain-core ];
 
+  propagatedBuildInputs = [ msgpack ];
+
+  pythonRelaxDeps = [ "msgpack" ]; # Can drop after 1.1.0 release
```
Comment (drupol) [inline pkgs/development/python-modules/langgraph-checkpoint/default.nix:38]: Is this comment still relevant now that version 2 is available?

Comment (sarahec) [reply pkgs/development/python-modules/langgraph-checkpoint/default.nix:38]: Sorry, I meant msgpack 1.10.0

Changing msgpack has to go through staging (2800+ affected)

Comment (sarahec) [reply pkgs/development/python-modules/langgraph-checkpoint/default.nix]: Should I comment around bedtime? Clearly not. In any case, I clarified the comment text. 

Thanks for catching it.

**Review by drupol — APPROVED** [review]

File: pkgs/development/python-modules/langgraph-checkpoint/default.nix (hunk 1 above)
Comment (drupol) [reply pkgs/development/python-modules/langgraph-checkpoint/default.nix]: Thanks 🙏

**Comment by natsukium** [comment]
@ofborg build python312Packages.langgraph python311Packages.langgraph

**Comment by sarahec** [comment]
@drupol looks like ofborg ate your approval

**Comment by drupol** [comment]
I don't understand why it's failing on the CI...

**Comment by sarahec** [comment]
> I don't understand why it's failing on the CI...

It was a bad hash on `langgraph`. That uncovered a flaky test that I had to disable.

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 346948` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>python311Packages.langgraph</li>
    <li>python311Packages.langgraph-checkpoint</li>
    <li>python311Packages.langgraph-checkpoint-postgres</li>
    <li>python311Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python311Packages.langgraph-checkpoint-sqlite</li>
    <li>python311Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python311Packages.langgraph-checkpoint.dist</li>
    <li>python311Packages.langgraph-sdk</li>
    <li>python311Packages.langgraph-sdk.dist</li>
    <li>python311Packages.langgraph.dist</li>
    <li>python312Packages.langgraph</li>
    <li>python312Packages.langgraph-checkpoint</li>
    <li>python312Packages.langgraph-checkpoint-postgres</li>
    <li>python312Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python312Packages.langgraph-checkpoint-sqlite</li>
    <li>python312Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python312Packages.langgraph-checkpoint.dist</li>
    <li>python312Packages.langgraph-sdk</li>
    <li>python312Packages.langgraph-sdk.dist</li>
    <li>python312Packages.langgraph.dist</li>
  </ul>
</details>

**Comment by sarahec** [comment]
@ofborg build python312Packages.langgraph python311Packages.langgraph

**Review by natsukium — APPROVED** [review]

**Pull request merged**